### PR TITLE
Run codecov in appveyor ci

### PR DESCRIPTION
**Pull request by atugushev**
Increases the test coverage.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -87,9 +89,13 @@ matrix:
...
+    - pip install tox codecov
 
 build: false
 
 test_script:
-    - C:\Python36\Scripts\tox
+    - tox
+
+after_test:
+    - codecov
```
Comment (blueyed) [inline .appveyor.yml]:
```suggestion
    - codecov --flags windows --name %TOXENV%
```

File: .appveyor.yml [hunk 2]
```diff
@@ -87,9 +89,13 @@ matrix:
     - PIP: master
 
 install:
-    - C:\Python36\python -m pip install tox
+    - "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
```
Comment (blueyed) [inline .appveyor.yml]:
```suggestion
    - "SET PATH=%PYTHON%\\Scripts;%PATH%"
```
Shouldn't that be enough?  (but just a guess)

**Comment by blueyed** [comment]
This will likely cause problems with codecov timing out though - if there are non already (given the huge number of builds for linux already).
We're having trouble (likely due to flags and number of builds) in pytest (and other projects) since months.

So in general it would be better to only run selective envs with coverage, by using a coverage factor in tox (see pytest's tox.ini).

File: .appveyor.yml (hunk 2 above)
Comment (atugushev) [reply .appveyor.yml]: The `python` binary in `C:\Python36\` folder. That's why `%PYTHON%` was added.

File: .appveyor.yml (hunk 1 above)
Comment (atugushev) [reply .appveyor.yml]: Good point. I think `--flag=linux` should be added also in the `.travis.yml`. What do you think?

Comment (blueyed) [reply .appveyor.yml]: Yes, in general - but flags might trigger timeouts with codecov.io, so I would rather only use name here for now.
It does not post comments anyway, does it?  (where the flags would be visible)

File: .appveyor.yml (hunk 2 above)
Comment (blueyed) [reply .appveyor.yml]: Ok.

**Comment by atugushev** [comment]
> We're having trouble (likely due to flags and number of builds) in pytest (and other projects) since months.

@blueyed could you please tell me what kind of problems do you mean?

**Comment by blueyed** [comment]
@atugushev 
When looking at PRs via codecov comments or the UI in general it might time out on their backend ("connection reset").
See https://github.com/pytest-dev/pytest/pull/4846#issuecomment-467949404.

**Comment by blueyed** [comment]
Here https://codecov.io/gh/jazzband/pip-tools/pull/745?src=pr&el=h1 is not timing out, although there are a lot of builds.
Source code size might also be a factor - I don't know.

**Comment by blueyed** [comment]
So I suggest adding flags ("windows" / "linux" (or rather `$TRAVIS_OS_NAME`)) here after all.

**Comment by atugushev** [comment]
@blueyed 
> So in general it would be better to only run selective envs with coverage, by using a coverage factor in tox (see pytest's tox.ini).
> See [pytest-dev/pytest#4846 (comment)](https://github.com/pytest-dev/pytest/pull/4846#issuecomment-467949404).

I see. Running selective envs with coverage sounds reasonable to me now. I'll try to test it and propose a PR soon.

> So I suggest adding flags ("windows" / "linux" (or rather `$TRAVIS_OS_NAME`)) here after all.

Builds section looks pretty good now, see the latest [codecov report](https://codecov.io/gh/jazzband/pip-tools/commit/1bc0bb195add121a2460cf91af1cc3551672afe0/build). Thanks for the suggestion, Daniel!

**Comment by blueyed** [comment]
For reference: extra coverage due to this: https://codecov.io/gh/jazzband/pip-tools/compare/c45862bd3e99df71fc020f828395fc7f6766be7c...fc69f3a2eab8d54c1a9850dcb2fdd9ec3d6930c5/changes

But it could be achieved with a single (or few) jobs only probably also.

**Pull request merged**